src/Interactive/Features/Interactive/Core/InteractiveHost.RemoteService.cs

```diff
@@ -74,9 +74,11 @@ internal void HookAutoRestartEvent()
                             _processExitHandlerStatus = ProcessExitHandlerStatus.Handled;
                             // Should set _processExitHandlerStatus before calling OnProcessExited to avoid deadlocks.
                             // Calling the host should be within the lock to prevent its disposing during the execution.
-                            await _host.OnProcessExited(Process).ConfigureAwait(false);
```

Comment:
> await _host.OnProcessExited(Process).ConfigureAwait(false); [](start = 27, length = 60)

Move outside of the lock and keep.

Reply:
`_host?.OnProcessExited`

---
In reply to: [188128872](https://github.com/dotnet/roslyn/pull/26770#discussion_r188128872) [](ancestors = 188128872)